src/client/src/ui/spotTile/SpotTileContainer.tsx

```diff
@@ -23,65 +22,25 @@ type SpotTileContainerStateProps = ReturnType<ReturnType<typeof makeMapStateToPr
 
 type SpotTileContainerProps = SpotTileContainerOwnProps & SpotTileContainerStateProps & SpotTileContainerDispatchProps
 
-class SpotTileContainer extends React.PureComponent<SpotTileContainerProps> {
-  componentDidMount() {
-    this.props.onMount()
-  }
-
-  render() {
-    const {
-      id,
-      currencyPair,
-      spotTileData,
-      onPopoutClick,
-      onNotificationDismissed,
-      tornOff,
-      displayCurrencyChart
-    } = this.props
-    if (!spotTileData || !spotTileData.price || !currencyPair) {
-      return null
-    }
-    return (
-      <TileSwitch
-        key={id}
-        currencyPair={currencyPair}
-        spotTileData={spotTileData}
-        onPopoutClick={onPopoutClick}
-        onNotificationDismissed={onNotificationDismissed}
-        executeTrade={this.executeTrade}
-        tornOff={tornOff}
-        displayCurrencyChart={displayCurrencyChart}
-      />
-    )
-  }
-
-  private executeTrade = (direction: Direction, notional: number) => {
-    const { executionConnected, spotTileData, currencyPair, executeTrade } = this.props
-    if (!executionConnected || spotTileData.isTradeExecutionInFlight || !spotTileData.price) {
-      return
-    }
-    const rate = direction === Direction.Buy ? spotTileData.price.ask : spotTileData.price.bid
-    const tradeRequestObj: TradeRequest = {
-      direction,
-      currencyBase: currencyPair.base,
-      symbol: currencyPair.symbol,
-      notional: notional || DEFAULT_NOTIONAL,
-      rawSpotRate: rate
-    }
-    executeTrade(createTradeRequest(tradeRequestObj))
-  }
-}
+const SpotTileContainer: React.SFC<SpotTileContainerProps> = ({ onMount, executionStatus, id, ...props }) => (
+  <Loadable
+    onMount={onMount}
+    status={executionStatus}
```

Comment:
I do not think this is right. When execution is down you can still see the application but you cannot trade.